pkg/controller/core/clusterqueue_controller.go

```diff
@@ -135,6 +158,10 @@ func (r *ClusterQueueReconciler) Update(e event.UpdateEvent) bool {
 	log := r.log.WithValues("clusterQueue", klog.KObj(cq))
 	log.V(2).Info("ClusterQueue update event")
 
+	if cq.DeletionTimestamp != nil {
```

Comment:
Like resizing the capacity of cq when cq is in termination.

Reply:
Yours two questions can be combined into one, currently, I won't handle the update event when a clusterQueue is in termination, like increase/decrease resources, flavors. For I haven't think it through yet. E.g. 
1. We have two cluster queues, cq1, cq2, they belongs to the same cohort1
2. Workload1 belongs to cq2 is pending for insufficient resources
3. cq1 is in termination
4. We upsize the resources of cq1 which leads to workloads1 admitted
5. After a while, cq1 is deleted. 

Then workload1 may stuck in starving resources.

Reply:
I don't think apiserver disallows it.

The case that @kerthcet mentions shouldn't happen because we should block the CQ for usage anyways. Then it doesn't really matter whether we update the internal cache or not.

So I'm fine with this.